$OpenBSD: patch-src_webserver_src_WebInterface_cpp,v 1.1 2011/07/26 15:32:36 dcoppa Exp $

Fix a crash when amuleweb is working as amuled child
(upstream git commit b0c3dab3fc005cb9e4fbbae20c0d0a42f2ffd9b6)

Use "litoral" as the default template

--- src/webserver/src/WebInterface.cpp.orig	Wed May  6 22:39:18 2009
+++ src/webserver/src/WebInterface.cpp	Tue Jul 26 16:47:37 2011
@@ -467,7 +467,7 @@ bool CamulewebApp::GetTemplateDir(const wxString& temp
 
 	
 	// template not found. reverting to default
-	const wxChar* const defaultTemplateName = wxT("php-default");
+	const wxChar* const defaultTemplateName = wxT("litoral");
 
 	if ( templateName == defaultTemplateName ) {
 		return false;
@@ -559,14 +559,15 @@ bool CamulewebApp::OnCmdLineParsed(wxCmdLineParser& pa
 		LoadAmuleConfig(cfg);
 		// do not process any other command-line parameters, use defaults instead
 
+		m_Verbose = false;
+		m_KeepQuiet = true;
+		m_LoadSettingsFromAmule = true;
+
 		if (!(m_TemplateOk = GetTemplateDir(m_TemplateName, m_TemplateDir))) {
 			// no reason to run webserver without a template
 			fprintf(stderr, "FATAL ERROR: Cannot find template: %s\n", (const char *)unicode2char(m_TemplateName));
-			return true;
+			return false;
 		}
-		m_Verbose = false;
-		m_KeepQuiet = true;
-		m_LoadSettingsFromAmule = true;
 		return true;
 	}
 
@@ -578,7 +579,7 @@ bool CamulewebApp::OnCmdLineParsed(wxCmdLineParser& pa
 
 		parser.Found(wxT("template"), &m_TemplateName);
 		if (m_TemplateName.IsEmpty()) {
-			m_TemplateName = wxT("php-default");
+			m_TemplateName = wxT("litoral");
 		}
 		if (!(m_TemplateOk = GetTemplateDir(m_TemplateName, m_TemplateDir))) {
 			// no reason to run webserver without a template
